lib/chef/knife/data_bag_secret_options.rb

```diff
@@ -98,41 +94,26 @@ def validate_secrets
       def base_encryption_secret_provided?(need_encrypt_flag = true)
         validate_secrets
 
-        return true if has_cl_secret? || has_cl_secret_file?
+        return true if config[:cl_secret] || config[:cl_secret_file]
 
         if need_encrypt_flag
           if config[:encrypt]
-            unless knife_config[:secret] || knife_config[:secret_file]
+            unless config[:secret] || config[:secret_file]
               ui.fatal("No secret or secret_file specified in config, unable to encrypt item.")
               exit(1)
             end
             return true
           end
           return false
-        elsif knife_config[:secret] || knife_config[:secret_file]
+        elsif config[:secret] || config[:secret_file]
           # Certain situations (show and bootstrap) don't need a --encrypt flag to use the config file secret
           return true
         end
         false
       end
 
-      def has_cl_secret?
-        Chef::Config[:knife].key?(:cl_secret)
-      end
-
-      def self.set_cl_secret(s)
-        Chef::Config[:knife][:cl_secret] = s
-      end
-
-      def has_cl_secret_file?
-        Chef::Config[:knife].key?(:cl_secret_file)
-      end
-
-      def self.set_cl_secret_file(sf)
-        Chef::Config[:knife][:cl_secret_file] = sf
-      end
-
       def knife_config
+        Chef.deprecated(:knife_bootstrap_apis, "The knife_config bootstrap helper has been deprecated, use the properly merged config helper instead")
```

Comment:
```suggestion
        Chef.deprecated(:knife_bootstrap_apis, "The `knife_config` bootstrap helper has been deprecated, use the properly merged `config` helper instead")
```

I could see this message being unclear if you weren't aware `config` referred to a method

Reply:
It pretty clearly suggests `knife_config` being rewritten to `config` as a solution though?

Reply:
After this is merged there needs to be a deprecation page added which describes it in a lot more words and gives examples and remediations, which should make it clearer.

Reply:
What I mean is if I'm unfamiliar with chef I could read "use the properly merged config helper instead" not realizing that `config` is a method that I could call. It reads like "instead of doing the bad thing, call the correct method instead".

Reply:
Still not quite getting it, think I'm too tired today.  What's your suggestion for a replacement?